### PR TITLE
Finish test on policy violation

### PR DESCRIPTION
**Pull request by natestuder**
Example bare metal policy violation handler.

### CODE REVIEW

**Comment by natestuder** [comment]
This was never merged when it was working previously.  It has since been resurrected, so can we get it in?

File: freedom-e-sdk/bsp/env/encoding.h [hunk 1]
```diff
@@ -872,6 +872,7 @@
 #define CAUSE_SUPERVISOR_ECALL 0x9
 #define CAUSE_HYPERVISOR_ECALL 0xa
 #define CAUSE_MACHINE_ECALL 0xb
+#define CAUSE_PIPE_VIOLATION 0x11
```
Comment (arunthomas) [inline freedom-e-sdk/bsp/env/encoding.h:875]: Update exception code to custom use code.

File: freedom-e-sdk/bsp/env/freedom-e300-hifive1/init.c [hunk 2]
```diff
@@ -185,6 +185,16 @@ static void uart_init(size_t baud_rate)
   UART0_REG(UART_REG_RXCTRL) |= UART_RXEN;
 }
 
+void handle_pipe_violation()
+{
+   const uint32_t fail = 0x3333;
```
Comment (arunthomas) [inline freedom-e-sdk/bsp/env/freedom-e300-hifive1/init.c:190]: #define for magic number

**Comment by natestuder** [comment]
This branch was merged manually.

**Pull request closed without merging**